sequencer/src/persistence/sql.rs

```diff
@@ -743,9 +764,9 @@ async fn collect_garbage(
         .map(|row| {
             let view: i64 = row.get("view");
             let leaf_data: Vec<u8> = row.get("leaf");
-            let leaf = bincode::deserialize::<Leaf>(&leaf_data)?;
+            let leaf = bincode::deserialize::<Leaf2>(&leaf_data)?;
```

Comment:
so we stored Leaf1 in anchor_leaf table before. We should deserialize it as Leaf1 and then convert it to Leaf2

Reply:
https://github.com/EspressoSystems/espresso-sequencer/pull/2344/commits/18c838e47b2d92d8c8870f8da393b16d486122bc

Reply:
Good catch,  thanks